symfit/core/minimizers.py

```diff
@@ -465,7 +466,76 @@ class Powell(ScipyMinimize, BaseMinimizer):
     """
     Wrapper around :func:`scipy.optimize.minimize`'s Powell algorithm.
     """
+    
 
+class TrustConstr(ScipyConstrainedMinimize, GradientMinimizer, BoundedMinimizer):
+    """
+    Wrapper around :func:`scipy.optimize.minimize`'s Trust-Constr algorithm.
+    """
+    def __init__(self, *args, **kwargs):
+        super(TrustConstr, self).__init__(*args, **kwargs)
+        # We have to break DRY because you cannot inherit from both
+        # ScipyConstrainedMinimize and ScipyGradientMinimize. So SLQSP is a
+        # special case. This is the same code as in ScipyGradientMinimize.
+        self.wrapped_jacobian = self.list2kwargs(self.wrapped_jacobian)
+
+    @keywordonly(jacobian=None, hess=None, options=None)
+    def execute(self, **minimize_options):
+        jacobian = minimize_options.pop('jacobian')
+        hess = minimize_options.pop('hess')
+        options = minimize_options.pop('options')
+        if options is None:
+            options = {}
+        # For models that are not differentiable, users need the ability to
```

Comment:
Good idea, but I think the logic pertaining to the jacobian should be moved to `ScipyGradientMinimize` so it can be used for any gradient method?